ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/kubernetes_cluster_deploy.py

```diff
@@ -2,144 +2,182 @@
+            """Please add your public ssh key, this will allow you to access the deployed container using ssh.
+                Just upload the ssh keys file with each key on a seperate line"""
         )
-        masternodes = form.int_ask(
-            "Please specify the number of master nodes", default=1, required=True, min=1, retry=retry
-        )  # minimum should be 1
-        workernodes = form.int_ask(
-            "Please specify the number of worker nodes", default=1, required=True, min=1, retry=retry
-        )  # minimum should be 1
-
-        form.ask()
-        cluster_size = sizes.index(cluster_size_string.value) + 1  # sizes are index 1
-        # Select nodes
-        if cluster_size == 1:
-            nodequery = {"sru": 50, "mru": 2, "cru": 1, "currency": currency}
-        else:
-            nodequery = {"sru": 100, "mru": 4, "cru": 2, "currency": currency}
-        try:
-            farms = j.sal.reservation_chatflow.farm_names_get(masternodes.value + workernodes.value, bot, **nodequery)
-            master_nodes_selected = j.sal.reservation_chatflow.nodes_get(
-                masternodes.value, farm_names=farms[: masternodes.value], **nodequery
+        self.ssh_keys_list = self.user_form_data["SSH keys"].split("\n")
+
+        self.user_form_data["Cluster secret"] = self.string_ask("Please add the cluster secret", default="secret")
+
+    @j.baseclasses.chatflow_step(title="Expiration")
+    def expiration_datetime(self):
+        self.expiration = self.datetime_picker(
+            "Please enter solution expiration time.",
+            required=True,
+            min_time=[3600, "Date/time should be at least 1 hour from now"],
+            default=j.data.time.epoch + 3900,
+        )
+        self.user_form_data["Solution expiration"] = j.data.time.secondsToHRDelta(self.expiration - j.data.time.epoch)
+
+    @j.baseclasses.chatflow_step(title="IP selection", disable_previous=True)
+    def ip_selection(self):
+
+        ipaddresses = list()
+        for idx, node_selected in enumerate(self.master_nodes_selected):
+            self.network.add_node(node_selected)
+            msg = f"Please choose IP Address for master node {idx + 1} of your kubernets cluster"
+            ip_address = self.network.ask_ip_from_node(node_selected, msg)
+            ipaddresses.append(ip_address)
+
+        for idx, node_selected in enumerate(self.worker_nodes_selected):
+            if node_selected not in self.master_nodes_selected:
+                self.network.add_node(node_selected)
+            msg = f"Please choose IP Address for worker node {idx + 1} of your kubernets cluster"
+            ip_address = self.network.ask_ip_from_node(node_selected, msg)
+            ipaddresses.append(ip_address)
+
+        self.user_form_data["IP Address"] = ipaddresses
+
+    @j.baseclasses.chatflow_step(title="Confirmation", disable_previous=True)
+    def overview(self):
+        self.md_show_confirm(self.user_form_data)
```

Comment:
the confirm itself should be in a seperate step cause what is the point of showing everything he selected if he can't go back?

Reply:
Fixed in all chatflows that had the confirmation disable going to previous step : 
https://github.com/threefoldtech/jumpscaleX_threebot/commit/7acbef7d8030b79316639179284a39c272825abe#diff-a5383166a7a1c1b6bd81d1638e7492c9L102